src/GlobalsAttributeReader.php

```diff
@@ -0,0 +1,113 @@
+<?php
+declare(strict_types=1);
+
+namespace Zalas\PHPUnit\Globals;
+
+use PHPUnit\Event\Code\TestMethod;
+use PHPUnit\Event\Test\Prepared;
+use PHPUnit\Event\Test\PreparedSubscriber;
+use Zalas\PHPUnit\Globals\Attribute\Env;
+use Zalas\PHPUnit\Globals\Attribute\Putenv;
+use Zalas\PHPUnit\Globals\Attribute\Server;
+
+final class GlobalsAttributeReader implements PreparedSubscriber
+{
+    public function notify(Prepared $event): void
+    {
+        $this->readGlobalAttributes($event->test());
+    }
+
+    private function readGlobalAttributes(TestMethod $method)
+    {
+        $setVars = $this->findSetVarAttributes($method);
+
+        foreach ($setVars as $var) {
+            match (true) {
+                $var instanceof Env => $_ENV[$var->name] = $var->value,
+                $var instanceof Server => $_SERVER[$var->name] = $var->value,
+                $var instanceof Putenv => \putenv(\sprintf('%s=%s', $var->name, $var->value))
+            };
+        }
+
+        $unsetVars = $this->findUnsetVarAttributes($method);
+
+        foreach ($unsetVars as $var) {
+            $callable = match (true) {
+                $var instanceof Env => static function($var) { unset($_ENV[$var->name]); },
```

Comment:
We need to wrap the unset part as `unset` is a keyword and can't be used inside a `match` expression.